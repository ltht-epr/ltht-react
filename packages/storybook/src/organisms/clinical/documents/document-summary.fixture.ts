import { DocumentReference, DocumentReferenceStatusCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetaData: Metadata = {
  dataSources: [],
  isRedacted: true,
  requestedWhen: '',
}

const Document1: DocumentReference = {
  id: '6b554095-bd5f-4134-8dbb-a21971959fd3',
  metadata: mockMetadata,
  content: [],
  description: 'Description of document one',
  indexed: {
    value: '2020-04-01T00:00:00+00:00',
  },
  created: {
    value: '2020-04-01T00:00:00+00:00',
  },
  status: DocumentReferenceStatusCode.Current,
  type: {},
}

const Document2: DocumentReference = {
  id: '2d889a55-d7ec-46ae-b674-23dfc6666014',
  metadata: mockMetadata,
  content: [],
  description: 'Description of document two',
  indexed: {
    value: '2019-10-11T00:00:00+00:00',
  },
  created: {
    value: '2019-10-11T00:00:00+00:00',
  },
  status: DocumentReferenceStatusCode.EnteredInError,
  type: {},
}

const Document3: DocumentReference = {
  id: '38962be0-4e5e-4640-a23e-af45da7c0578',
  metadata: redactedMetaData,
  content: [],
  description: 'Description of document three',
  indexed: {
    value: '2020-01-23T00:00:00+00:00',
  },
  created: {
    value: '2020-01-11T00:00:00+00:00',
  },
  status: DocumentReferenceStatusCode.Superseded,
  type: {},
}

const Document4: DocumentReference = {
  id: '38962be0-4e5e-4640-a23e-af45da7c0777',
  metadata: mockMetadata,
  content: [],
  description: 'Description of document three',
  indexed: {
    value: '2020-01-23T00:00:00+00:00',
  },
  created: {
    value: '2020-01-23T00:00:00+00:00',
  },
  status: DocumentReferenceStatusCode.Superseded,
  type: {},
}

const documents: DocumentReference[] = [Document1, Document2, Document3, Document4]

export default documents
