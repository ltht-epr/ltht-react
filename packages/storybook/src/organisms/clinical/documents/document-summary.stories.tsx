import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import DocumentSummary from '@ltht-react/document-summary'
import readme from '@ltht-react/document-summary/README.md'
import Card from '@ltht-react/card'
import documents from './document-summary.fixture'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const stories = storiesOf('Organisms - Clinical|Document', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Summary', () => {
  return (
    <Card>
      <Card.Header>
        <Card.Title>Documents</Card.Title>
      </Card.Header>
      <Card.List>
        {documents.map(document => (
          <Card.ListItem key={document.id}>
            <DocumentSummary document={document} />
          </Card.ListItem>
        ))}
      </Card.List>
    </Card>
  )
})
