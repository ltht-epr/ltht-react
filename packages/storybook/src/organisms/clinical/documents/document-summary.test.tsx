import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import DocumentSummary from '@ltht-react/document-summary'
import documents from './document-summary.fixture'

const Sut: React.FC = () => {
  return (
    <>
      {documents.map(document => (
        <div key={document.id}>
          <DocumentSummary document={document} />
        </div>
      ))}
    </>
  )
}

describe('DocumentSummaryy', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Sut />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Sut />)).toMatchSnapshot('wrapper mount')
  })
})
