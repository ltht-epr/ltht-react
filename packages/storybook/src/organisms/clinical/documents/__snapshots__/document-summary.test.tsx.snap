// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentSummaryy matches snapshot: wrapper mount 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 0.5rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: right;
}

.emotion-2 {
  color: #768692;
}

.emotion-10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-12 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: right;
}

.emotion-11 {
  color: #768692;
}

<Sut>
  <div
    key="6b554095-bd5f-4134-8dbb-a21971959fd3"
  >
    <DocumentSummary
      document={
        Object {
          "content": Array [],
          "created": Object {
            "value": "2020-04-01T00:00:00+00:00",
          },
          "description": "Description of document one",
          "id": "6b554095-bd5f-4134-8dbb-a21971959fd3",
          "indexed": Object {
            "value": "2020-04-01T00:00:00+00:00",
          },
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "status": "CURRENT",
          "type": Object {},
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <Styled(div)>
            <div
              className="emotion-0"
            >
              <DocumentDate
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "description": "Description of document one",
                    "id": "6b554095-bd5f-4134-8dbb-a21971959fd3",
                    "indexed": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "CURRENT",
                    "type": Object {},
                  }
                }
              >
                <div>
                  Apr-01,-2020
                </div>
              </DocumentDate>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-1"
            >
              <DocumentDescription
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "description": "Description of document one",
                    "id": "6b554095-bd5f-4134-8dbb-a21971959fd3",
                    "indexed": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "CURRENT",
                    "type": Object {},
                  }
                }
              >
                Description of document one
              </DocumentDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-3"
            >
              <DocumentSource
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "description": "Description of document one",
                    "id": "6b554095-bd5f-4134-8dbb-a21971959fd3",
                    "indexed": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "CURRENT",
                    "type": Object {},
                  }
                }
              >
                <div>
                  Mock
                </div>
              </DocumentSource>
              <DocumentStatus
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "description": "Description of document one",
                    "id": "6b554095-bd5f-4134-8dbb-a21971959fd3",
                    "indexed": Object {
                      "value": "2020-04-01T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "CURRENT",
                    "type": Object {},
                  }
                }
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    Current
                  </div>
                </Styled(div)>
              </DocumentStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </DocumentSummary>
  </div>
  <div
    key="2d889a55-d7ec-46ae-b674-23dfc6666014"
  >
    <DocumentSummary
      document={
        Object {
          "content": Array [],
          "created": Object {
            "value": "2019-10-11T00:00:00+00:00",
          },
          "description": "Description of document two",
          "id": "2d889a55-d7ec-46ae-b674-23dfc6666014",
          "indexed": Object {
            "value": "2019-10-11T00:00:00+00:00",
          },
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "status": "ENTERED_IN_ERROR",
          "type": Object {},
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <Styled(div)>
            <div
              className="emotion-0"
            >
              <DocumentDate
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "description": "Description of document two",
                    "id": "2d889a55-d7ec-46ae-b674-23dfc6666014",
                    "indexed": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "ENTERED_IN_ERROR",
                    "type": Object {},
                  }
                }
              >
                <div>
                  Oct-11,-2019
                </div>
              </DocumentDate>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-1"
            >
              <DocumentDescription
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "description": "Description of document two",
                    "id": "2d889a55-d7ec-46ae-b674-23dfc6666014",
                    "indexed": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "ENTERED_IN_ERROR",
                    "type": Object {},
                  }
                }
              >
                Description of document two
              </DocumentDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-3"
            >
              <DocumentSource
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "description": "Description of document two",
                    "id": "2d889a55-d7ec-46ae-b674-23dfc6666014",
                    "indexed": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "ENTERED_IN_ERROR",
                    "type": Object {},
                  }
                }
              >
                <div>
                  Mock
                </div>
              </DocumentSource>
              <DocumentStatus
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "description": "Description of document two",
                    "id": "2d889a55-d7ec-46ae-b674-23dfc6666014",
                    "indexed": Object {
                      "value": "2019-10-11T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "ENTERED_IN_ERROR",
                    "type": Object {},
                  }
                }
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    Entered In Error
                  </div>
                </Styled(div)>
              </DocumentStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </DocumentSummary>
  </div>
  <div
    key="38962be0-4e5e-4640-a23e-af45da7c0578"
  >
    <DocumentSummary
      document={
        Object {
          "content": Array [],
          "created": Object {
            "value": "2020-01-11T00:00:00+00:00",
          },
          "description": "Description of document three",
          "id": "38962be0-4e5e-4640-a23e-af45da7c0578",
          "indexed": Object {
            "value": "2020-01-23T00:00:00+00:00",
          },
          "metadata": Object {
            "dataSources": Array [],
            "isRedacted": true,
            "requestedWhen": "",
          },
          "status": "SUPERSEDED",
          "type": Object {},
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <DocumentRedacted
            document={
              Object {
                "content": Array [],
                "created": Object {
                  "value": "2020-01-11T00:00:00+00:00",
                },
                "description": "Description of document three",
                "id": "38962be0-4e5e-4640-a23e-af45da7c0578",
                "indexed": Object {
                  "value": "2020-01-23T00:00:00+00:00",
                },
                "metadata": Object {
                  "dataSources": Array [],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "status": "SUPERSEDED",
                "type": Object {},
              }
            }
          >
            <Styled(div)>
              <div
                className="emotion-10"
              >
                <DocumentDate
                  document={
                    Object {
                      "content": Array [],
                      "created": Object {
                        "value": "2020-01-11T00:00:00+00:00",
                      },
                      "description": "Description of document three",
                      "id": "38962be0-4e5e-4640-a23e-af45da7c0578",
                      "indexed": Object {
                        "value": "2020-01-23T00:00:00+00:00",
                      },
                      "metadata": Object {
                        "dataSources": Array [],
                        "isRedacted": true,
                        "requestedWhen": "",
                      },
                      "status": "SUPERSEDED",
                      "type": Object {},
                    }
                  }
                >
                  <div>
                    Jan-11,-2020
                  </div>
                </DocumentDate>
              </div>
            </Styled(div)>
            <Styled(div)>
              <div
                className="emotion-12"
              >
                <RedactedDescription>
                  <div
                    className="emotion-11"
                  >
                    Insufficient privileges
                  </div>
                </RedactedDescription>
              </div>
            </Styled(div)>
          </DocumentRedacted>
        </div>
      </Styled(div)>
    </DocumentSummary>
  </div>
  <div
    key="38962be0-4e5e-4640-a23e-af45da7c0777"
  >
    <DocumentSummary
      document={
        Object {
          "content": Array [],
          "created": Object {
            "value": "2020-01-23T00:00:00+00:00",
          },
          "description": "Description of document three",
          "id": "38962be0-4e5e-4640-a23e-af45da7c0777",
          "indexed": Object {
            "value": "2020-01-23T00:00:00+00:00",
          },
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "status": "SUPERSEDED",
          "type": Object {},
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <Styled(div)>
            <div
              className="emotion-0"
            >
              <DocumentDate
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "description": "Description of document three",
                    "id": "38962be0-4e5e-4640-a23e-af45da7c0777",
                    "indexed": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "SUPERSEDED",
                    "type": Object {},
                  }
                }
              >
                <div>
                  Jan-23,-2020
                </div>
              </DocumentDate>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-1"
            >
              <DocumentDescription
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "description": "Description of document three",
                    "id": "38962be0-4e5e-4640-a23e-af45da7c0777",
                    "indexed": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "SUPERSEDED",
                    "type": Object {},
                  }
                }
              >
                Description of document three
              </DocumentDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-3"
            >
              <DocumentSource
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "description": "Description of document three",
                    "id": "38962be0-4e5e-4640-a23e-af45da7c0777",
                    "indexed": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "SUPERSEDED",
                    "type": Object {},
                  }
                }
              >
                <div>
                  Mock
                </div>
              </DocumentSource>
              <DocumentStatus
                document={
                  Object {
                    "content": Array [],
                    "created": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "description": "Description of document three",
                    "id": "38962be0-4e5e-4640-a23e-af45da7c0777",
                    "indexed": Object {
                      "value": "2020-01-23T00:00:00+00:00",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "status": "SUPERSEDED",
                    "type": Object {},
                  }
                }
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    Superseded
                  </div>
                </Styled(div)>
              </DocumentStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </DocumentSummary>
  </div>
</Sut>
`;
